packages/angular-cli/tasks/extract-i18n.ts

```diff
@@ -0,0 +1,60 @@
+const Task = require('../ember-cli/lib/models/task');
+
+import * as compiler from '@angular/compiler';
+import {Extractor} from '@angular/compiler-cli';
+import * as tsc from '@angular/tsc-wrapped';
+import * as ts from 'typescript';
+import * as path from 'path';
+
+export const Extracti18nTask = Task.extend({
+  run: function () {
+    const project = path.resolve(this.project.root, 'src');
+    const cliOptions = new tsc.I18nExtractionCliOptions({
+      i18nFormat: this.i18nFormat
+    });
+
+    function extract (
+      ngOptions: tsc.AngularCompilerOptions, cliOptions: tsc.I18nExtractionCliOptions,
+      program: ts.Program, host: ts.CompilerHost) {
+
+      const resourceLoader: compiler.ResourceLoader = {
+        get: (s: string) => {
+          if (!host.fileExists(s)) {
+            // Return empty string to avoid extractor stop processing
+            return Promise.resolve('');
```

Comment:
Would it be useful to compile less/sass files? Also, please note that in the future we will support other format, even for templates (pug comes to mind).

Are you available on gitter right now?  If so, contact me and we can review this interactively for fast tracking.